lowranceusr.cc

```diff
@@ -1,238 +1,474 @@
-	it under the terms of the GNU General Public License as published by
-	the Free Software Foundation; either version 2 of the License, or
-	(at your option) any later version.
-
-	This program is distributed in the hope that it will be useful,
-	but WITHOUT ANY WARRANTY; without even the implied warranty of
-	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-	GNU General Public License for more details.
-
-	You should have received a copy of the GNU General Public License
-	along with this program; if not, write to the Free Software
-	Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111 USA
-
-	HISTORY:
-
-	6/21/05 - Ling Nero (rnlnero@yahoo.com)
-	- Added Routes, Icons, & Tracks support
-	- Fixed waypoint date/time stamp conversion
-	02/09/08 - oliskoli
-	- gbfile API
-	- check for buffer overflows when reading names or comments
-	02/25/2008 - Alan Porter (alan@kr4jb.net)
-	- Added new icons for Lowrance iFinder Expedition C
-	- Categorized geocaching waypoints using different icons
+  Access to Lowrance USR files.
+
+  Initial contribution to gpsbabel by Jason Rust (jrust at rustyparts.com)
+
+  Copyright (C) 2005 - 2018 Robert Lipe, robertlipe+source@gpsbabel.org
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111 USA
+
+  HISTORY:
+
+  6/21/05 - Ling Nero (rnlnero@yahoo.com)
+    - Added Routes, Icons, & Tracks support
+    - Fixed waypoint date/time stamp conversion
+  02/09/08 - oliskoli
+    - gbfile API
+    - check for buffer overflows when reading names or comments
+  02/25/2008 - Alan Porter (alan@kr4jb.net)
+    - Added new icons for Lowrance iFinder Expedition C
+    - Categorized geocaching waypoints using different icons
+  01/06/2012 - Kris Beevers (beevek at gmail.com)
+    - First pass read-write support for USR 4 in lowranceusr4.cc
+  10/02/2018 - BJ Kowalski (bj dot kowalski at gmail.com)
+    - Added support to lowranceusr4.cc for USR 4 route version 2
+      format data.
+    - Combined processing of USR 2/3/4/5/6 formats into a single
+      file. The previous implementation did not support conversion
+      between USR 2/3 and 4/5/6 or between 4/5/6 and 2/3 beause of 
+      the seperation in processing between lowranceusr.cc and 
+      lowranceusr4.cc.
+
+  USR Background Information
+
+  Collected from various WEB sources and Lowrance HOOK2 User Manual.
```

Comment:
For maintenance it is very nice to have the sources of information available.  Can you list the URLs?

Reply:
Added URLs per request.